src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -23,75 +26,79 @@ const useMessagePreviews = () => {
   const setPreviewMessages = useMessageStore((state) => state.setPreviewMessages);
   const [messagesLoading, setMessagesLoading] = useState<boolean>();
 
-  const peerAddresses = Array.from(conversations.keys());
-  const request = { ownedBy: peerAddresses, limit: 50 };
-  const {
-    loading: profilesLoading,
-    error: profilesError,
-    fetchMore
-  } = useQuery(ProfilesDocument, {
+  const profileIds = Array.from(conversations.keys())
+    .map((conversationKey: string) => {
+      const match = parseConversationKey(conversationKey);
+      return (
+        match && match.members.find((member) => member && currentProfile && member !== currentProfile.id)
+      );
+    })
+    .filter((profileId) => !!profileId);
+
+  const request = { profileIds: profileIds };
```

Comment:
I'm not sure I fully understand the motivation to query by profile ids here instead of addresses so I ran thru a profile transferred scenario and here's what I'm thinking:

- We're messaging with a person at ETH address 0x123 who owns profile A & B, and we're talking with A.
- 0x123 transfers profile A to ETH address 0x789.
- We query convos and get our convo history with `0x123-A` back.
- Option 1: We query for profiles by id (A) which now has 0x789 as the owner: `0x789-A` is the profile key.
- Option 2: We query for profiles by address (0x123) which returns profile B: `0x123-B` is the profile key.

Neither option 1 or option 2's new profile key will match a message with the original key `0x123-A` so neither should render as-is right now. Also in either scenario, I think we could look at all profile keys that don't match messages but have either the same profile id (option 1) or eth address (option 2) in the key, and render those convos as locked.

Couldn't querying profiles by addresses still work? Or am I missing something behind the motivation to use profile ids? I think I'm partial to using addresses to stitch xmtp <> lens data together since it's the only constant, but would love to hear your thoughts!

Reply:
One advantage to querying by profile IDs is that we only get back the profiles we actually care about. If someone happens to own 20, it won't bloat the list of returned results when we only care about 1. 

But yes, either way is possible. We could totally query by address and then zip back together the correct profile:convo pairs from the results and throw out the mismatches.

Reply:
Alright cool, makes sense! I'm good with profileIds then. Thanks for helping walk me thru it. Feel like I understand the flow now.